src/helpers.js

```diff
@@ -25,9 +25,9 @@ function _runWithRealTimers(callback) {
 
   const callbackReturnValue = callback()
 
-  const usedJestFakeTimers = Object.entries(timerAPI).some(
-    ([name, func]) => func !== globalObj[name],
-  )
+  const usedJestFakeTimers =
+    typeof jest !== 'undefined' &&
```

Comment:
At the cost of having multiple return points though. Not sure this would actually improve things.

Reply:
Might have a better idea.

Reply:
I like the early return but not in this function. I moved the check higher up the tree so that we now have dedicated functions for jest timers and framework agnostic timer functions. This also gets rid of the weird `runWithRealTimers` vs `_runWithRealTimers` where `_runWithRealTimers` is now `runWithJestRealTimers`

Reply:
@timdeschryver What do you think about [`a29bd91` (#900)](https://github.com/testing-library/dom-testing-library/pull/900/commits/a29bd919b3815912f76a1cb95200d43c8292f154)? The branching makes more sense to me now.

Reply:
I like it! 👍